addons/web/static/tests/legacy/fields/basic_fields_tests.js

```diff
@@ -769,6 +769,40 @@ QUnit.module('basic_fields', {
         form.destroy();
     });
 
+    QUnit.test('numeric field: field with type `number` and with keydown on numpad decimal key', async function (assert) {
+        patchWithCleanup(basicFields.NumericField.constructor.prototype, {_onKeydown(ev) {
+            const res = this._super(...arguments);
+            assert.step("Using the numpad decimal did not crash");
+            return res;
+            }})
```

Comment:
I realized that the important assertion we can make here is that the browser will apply the default behavior of the keydown event (also a bit of nitpicking: code style)
```suggestion
        assert.expect(2); // (as we don't use verifySteps anymore)

        patchWithCleanup(basicFields.NumericField.constructor.prototype, {
           _onKeydown(ev) {
                const res = this._super(...arguments);

                // This _onKeydown handler must not prevent default
                // a keydown event for NumericField with type=number
                assert.ok(!ev.defaultPrevented);

                return res;
            },
        });
```